client/my-sites/promote-post/main.tsx

```diff
@@ -81,6 +94,21 @@ export default function PromotedPosts( { tab }: Props ) {
 		isRequestingPostsForQuery( state, selectedSiteId, queryProducts )
 	);
 
+	const topViewedPostAndPages = useSelector( ( state ) =>
+		getTopPostAndPages( state, selectedSiteId, topPostsQuery )
+	);
+
+	const mostPopularPostAndPages = useSelector( ( state ) => {
+		const topPosts: any[ PostType ] = [];
+
+		topViewedPostAndPages?.map( ( post: any ) => {
```

Comment:
`map` is mostly used when we need to return an array; in this case, `forEach` might be a better option

Reply:
[Improved it here](https://github.com/Automattic/wp-calypso/pull/73760/commits/520a576acf5ba98efa46ef9880e227110fe24e6f).